TinCanTests/ActivityTest.cs

```diff
@@ -40,12 +40,16 @@ public void TestActivityIdCase()
         }
 
         [Test]
-        [ExpectedException("System.UriFormatException")]
         public void TestActivityIdInvalidUri()
         {
-            var activity = new Activity();
-            string invalid = "foo";
-            activity.id = invalid;
+            Assert.Throws<System.UriFormatException>(
```

Comment:
https://github.com/nunit/docs/wiki/Assertions

> In NUnit 3.0, assertions are written primarily using the `Assert.That` method, which takes constraint objects as an argument. We call this the Constraint Model of assertions.

> In earlier versions of NUnit, a separate method of the Assert class was used for each different assertion. This Classic Model is still supported but since no new features have been added to it for some time, the constraint-based model must be used in order to have full access to NUnit's capabilities.

I mean, the answer to this question is nowhere near as important as "do we have good tests." But the docs seem pretty clear that NUnit developers are trying to move the NUnit community to `Assert.That`. It's as an explicit a statement as we can get, and it's really the only data point we have about what the NUnit community standards actually are. As a veteran of the Python 2/3 wars, community standards are something I'm kind of sensitive about.

I think the claim that "`Assert.That` is more generic and `Assert.Throws` is more specific" is incredibly misleading, because it's not about `Assert.That` versus `Assert.Throws`, it's about `Assert.That` versus the whole classic model, and  `Assert.That(xyz, Throws...)` versus `Assert.Throws`.

If I had to sell you the value of the semantic difference, it's more that `Assert.That` allows you to write more more complicated asserts when you want to extend tests. With the constraint model, it's easier to say `Assert.That(aFunction, Throws.InstanceOf<UriFormatException>().With.Message.Containing("activity ID"))`. (Such a thing is absolutely essential, I think, if your function could throw the same type of exception for different reasons.) You can have the same sorts of tests with Assert.Throws by having the exception returned and then checking the properties, but I think the constraint model is far more readable here: "Assert that a function throws an instance of a UriFormatException with a message that contains the string 'activity ID'."